sdk/formrecognizer/Azure.AI.FormRecognizer/src/Generated/Models/ServiceResourceDetails.cs

```diff
@@ -15,14 +15,20 @@ internal partial class ServiceResourceDetails
     {
         /// <summary> Initializes a new instance of ServiceResourceDetails. </summary>
         /// <param name="customDocumentModels"> Details regarding custom document models. </param>
-        /// <param name="customNeuralDocumentModelBuilds"> Quota used, limit, and next reset date/time. </param>
-        /// <exception cref="ArgumentNullException"> <paramref name="customDocumentModels"/> or <paramref name="customNeuralDocumentModelBuilds"/> is null. </exception>
-        internal ServiceResourceDetails(CustomDocumentModelsDetails customDocumentModels, QuotaDetails customNeuralDocumentModelBuilds)
```

Comment:
I wonder what's the right pattern to follow in these cases. For example, should we add this one back as a custom convenience constructor and have both constructors? If we did, should that constructor keep the `Argument.AssetNotNull` check?

Reply:
We already have a generated constructor that takes all parameters so we can't add it back. What we have today, after these PR changes is:
```cs
// Only takes required properties. Null checks everything.
// In theory, used to instantiate this object in manually written code, but we don't use it anywhere (0 references).
internal ServiceResourceDetails(CustomDocumentModelsDetails customDocumentModels) {}

// Takes all properties. Doesn't make any checks.
// Used by our generated deserialization method.
internal ServiceResourceDetails(CustomDocumentModelsDetails customDocumentModels, QuotaDetails customNeuralDocumentModelBuilds) {}
```

However, before this PR, all properties were required, so we couldn't have both constructors. This is what we had:
```cs
// Only takes required properties (all of them). Null checks everything.
// In theory, used to instantiate this object in manually written code, but also used by our generated deserialization method.
internal ServiceResourceDetails(CustomDocumentModelsDetails customDocumentModels, QuotaDetails customNeuralDocumentModelBuilds) {}
```

Since now we have at least one optional property, as new required properties are added, CodeGen will still pick the constructor that makes no checks.